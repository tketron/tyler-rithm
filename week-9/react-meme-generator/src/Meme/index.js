import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

function Meme(props) {
  return (
    <div className="Meme" id="foo">
      <div className="container">
        <span className="top-text">{props.topText}</span>
        <img src={props.url} alt="a meme" />
        <span className="bottom-text">{props.bottomText}</span>
        <button id="meme_deleteBtn" onClick={() => props.deleteMeme(props.id)}>
          DELETE
        </button>
      </div>
    </div>
  );
}

Meme.propTypes = {
  topText: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  bottomText: PropTypes.string.isRequired
};

export default Meme;
