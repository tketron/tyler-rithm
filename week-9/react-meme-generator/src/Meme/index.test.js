import React from 'react'; // for JSX
import { shallow } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import Meme from '.'; // import the component itself

describe('<Meme />', () => {
  let wrapper;
  it('renders', () => {
    // smoke test!!!
    wrapper = shallow(
      <Meme
        url="https://i.ytimg.com/vi/I7jgu-8scIA/maxresdefault.jpg"
        topText="I can haz"
        bottomText="react tests"
      />
    );
  });
  it('matches snapshot', () => {
    //snapshot test
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
