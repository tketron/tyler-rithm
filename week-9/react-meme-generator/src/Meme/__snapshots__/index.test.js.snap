// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Meme /> matches snapshot 1`] = `
<div
  className="Meme"
  id="foo"
>
  <div
    className="container"
  >
    <span
      className="top-text"
    >
      I can haz
    </span>
    <img
      alt="a meme"
      src="https://i.ytimg.com/vi/I7jgu-8scIA/maxresdefault.jpg"
    />
    <span
      className="bottom-text"
    >
      react tests
    </span>
    <button
      id="meme_deleteBtn"
      onClick={[Function]}
    >
      DELETE
    </button>
  </div>
</div>
`;
